tools/registrar-scraper/src/index.ts

```diff
@@ -52,6 +52,61 @@ async function main() {
       data: Object.values(instructorInfo),
     }),
   ]);
+  const courses = Object.fromEntries((await prisma.course.findMany()).map((x) => [x.id, x]));
+  const instructors = Object.fromEntries(
+    (await prisma.instructor.findMany()).map((x) => [x.ucinetid, x]),
+  );
+  const newCourses = Object.values(courses).map((course) => ({
+    ...course,
+    prerequisiteTree: course.prerequisiteTree as object,
+    instructors: course.instructorHistory
+      .map((x) => instructors[x])
+      .filter((x) => x)
```

Comment:
Slightly more efficient because we don't have to allocate the filter function multiple times, and a bit more legible because it's clear we're filtering for non-falsy values.

```suggestion
      .filter(notNull)
```